lib/gds_api/publishing_api_v2.rb

```diff
@@ -324,6 +324,19 @@ def get_editions(params = {})
     get_json(get_editions_url(params))
   end
 
+  def get_paged_editions(params = {})
+    Enumerator.new do |yielder|
+      next_link = get_editions_url(params)
+      while next_link
+        yielder.yield begin
+          response = get_json(next_link)
```

Comment:
I'd originally thought about doing that, but it felt out-of-place in `gds-api-adapters`, which has all(?) of its methods return a `Response` object. It felt more natural to return pages of responses, rather than actual results, and then leave the clients to do things like `response.map { |response| response['results'] }.flatten` or similar.

For example, what if `publishing-api` errors? Is that really the responsibility of `gds-api-adapters`?